src/main/java/core/basesyntax/model/FruitTransaction.java

```
@@ -0,0 +1,71 @@
+package core.basesyntax.model;
+
+import java.util.Arrays;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public void setOperation(Operation operation) {
+        this.operation = operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(String fruit) {
+        this.fruit = fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    @Override
+    public String toString() {
+        return "FruitTransaction{"
+                + "operation=" + operation
+                + ", fruit='" + fruit + '\''
+                + ", quantity=" + quantity + '}';
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private String letter;
+
+        Operation(String operation) {
+            this.letter = operation;
+        }
+
+        public String getLetter() {
+            return letter;
+        }
+
+        public static FruitTransaction.Operation getTypeOperation(String letter) {
+            return Arrays.stream(FruitTransaction.Operation.values())
+                    .filter(o -> o.getLetter().equals(letter))
+                    .findFirst()
+                    .get();
```

let's throw an exception in case if letter isn't valid